src/status.js

```diff
@@ -1,38 +1,116 @@
+    databaseUrl: credStatusDatabaseUrl,
+    databaseHost: credStatusDatabaseHost,
+    databasePort: credStatusDatabasePort,
+    databaseUsername: credStatusDatabaseUsername,
+    databasePassword: credStatusDatabasePassword,
+    didMethod: 'key',
+    didSeed: credStatusDidSeed,
+    signUserCredential: false,
+    signStatusCredential: true
+  });
+}
+
+async function createGitHubStatusManager() {
+  return createStatusManagerGit({
+    service: credStatusService,
+    repoName: credStatusRepoName,
+    metaRepoName: credStatusMetaRepoName,
+    ownerAccountName: credStatusOwnerAccountName,
+    repoAccessToken: credStatusAccessToken,
+    metaRepoAccessToken: credStatusAccessToken,
+    didMethod: 'key',
+    didSeed: credStatusDidSeed,
+    signUserCredential: false,
+    signStatusCredential: true
+  });
+}
+
+async function createGitLabStatusManager() {
+  return createStatusManagerGit({
+    service: credStatusService,
+    repoName: credStatusRepoName,
+    repoId: credStatusRepoId,
+    metaRepoName: credStatusMetaRepoName,
+    metaRepoId: credStatusMetaRepoId,
+    ownerAccountName: credStatusOwnerAccountName,
+    repoAccessToken: credStatusAccessToken,
+    metaRepoAccessToken: credStatusAccessToken,
+    didMethod: 'key',
+    didSeed: credStatusDidSeed,
+    signUserCredential: false,
+    signStatusCredential: true
+  });
+}
+
 /* we allow passing in a status manager, for testing */
 async function initializeStatusManager(statusManager) {
-    if (statusManager) {
-        STATUS_LIST_MANAGER = statusManager
-    } else if (!STATUS_LIST_MANAGER) {
-            STATUS_LIST_MANAGER = await createStatusManager({
-                service: 'github',
-                repoName: credStatusRepoName,
-                metaRepoName: credStatusMetaRepoName,
-                ownerAccountName: credStatusOwnerAccountName,
-                repoAccessToken: credStatusAccessToken,
-                metaRepoAccessToken: credStatusAccessToken,
-                didMethod: 'key',
-                didSeed: credStatusDidSeed,
-                signUserCredential: false,
-                signStatusCredential: true
-            });
-    }
+  if (statusManager) {
+    STATUS_LIST_MANAGER = statusManager;
+    return;
+  } else if (STATUS_LIST_MANAGER) {
+    return;
+  }
+
+  switch (credStatusService) {
+    case 'mongodb':
+      STATUS_LIST_MANAGER = await createDatabaseStatusManager();
+      break;
+    case 'github':
+      STATUS_LIST_MANAGER = await createGitHubStatusManager();
+      break;
+    case 'gitlab':
+      STATUS_LIST_MANAGER = await createGitLabStatusManager();
+      break;
+    default:
+      throw new Error('Encountered unsupported credential status service');
+  }
+}
+
+async function getStatusManager() {
+  await initializeStatusManager();
+  return STATUS_LIST_MANAGER;
 }
 
-function getStatusManager() {
-    return STATUS_LIST_MANAGER;
+async function getStatusCredential(statusCredentialId) {
```

Comment:
I wonder if there'd be any value in having the git status managers also return the status credential for this call.  You could imagine someone wanting to use git, but maybe also wanting to use a redirect from some other DNS entry, so that they could later change their status implementation (or even just which GitHub repo is used) without breaking their past issued credentials.

Reply:
Hmm, I suppose it couldn't hurt. The main reason why I excluded the git status manager is because it isn't required for the status service to host status credentials, since those are hosted in the Git service. But if we would like to enable that as a convenience, I can update the Git status manager to implement the `getStatusCredential` method.